client/dashboard/app/auth/index.tsx

```diff
@@ -75,14 +77,29 @@ export function AuthProvider( { children }: { children: React.ReactNode } ) {
 		};
 	}, [ user ] );
 
-	if ( userIsError ) {
-		if ( typeof window !== 'undefined' ) {
-			const currentPath = window.location.pathname;
-			const loginUrl = `/log-in?redirect_to=${ encodeURIComponent( currentPath ) }`;
-			window.location.href = loginUrl;
-		}
-		return null;
-	}
+	// Subscribe to network errors and when errors occur due to being logged
+	// out, redirect the user to the log in screen.
+	useEffect( () => {
+		const handleEvent = ( event: MutationCacheNotifyEvent | QueryCacheNotifyEvent ) => {
+			if (
+				event.type === 'updated' &&
+				event.action.type === 'error' &&
+				isWpError( event.action.error ) &&
+				event.action.error.statusCode === 403 &&
+				event.action.error.error === 'authorization_required'
```

Comment:
Does this mirror HTTP code errors? Since 403 is forbidden and 401 is unauthorized 🤔

Reply:
Yeah the mismatch between the naming of status and error codes is unfortunate, but this is what the APIs actually return.